packages/lesswrong/server/register-mui-styles/index.ts

```diff
@@ -0,0 +1,154 @@
+// Import to add MUI components to our component registry.
+//
+// This is done purely for the side effect of adding their styles
+// to the allStyles.css sheet generated from registered components,
+// overriding default styles.
+//
+// This is desirable because it will be easier to support light/dark mode
+// with allStyles.css -- see note in server/.../renderPage.tsx.
+//
+// The `styles` callbacks should return the subsets of built-in styles which
+// reference the theme's palette. A utility ./scripts/gen-mui-registration.sh
+// is provided to help generate these. Example usage:
+//
+//   echo 'Paper Button' | bash ./scripts/gen-mui-registration.sh
+//
+// The script uses the material-ui source to get the original styles,
+// which aren't accessible at runtime. It assumes that you have
+// https://github.com/mui/material-ui checked out at branch v3.x and
+// `kak`, the kakoune text editor (used for string processing), somewhere
+// on your PATH.
+//
+// The script is a super hacky regex-based thing and you should double
+// check it's output.
+//
+// Also, I wouldn't run it with untrusted input.
+
+// TODO: Add registerStyles() to register styles without the component.
+import { registerComponent } from '../../lib/vulcan-lib';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+
+registerComponent("MuiPaper", Paper, { styles: theme => {
+  return {
+    root: {
+      backgroundColor: theme.palette.background.paper,
+    },
+  };
+}});
+
+registerComponent("MuiButton", Button, { styles: theme => ({
```

Comment:
Compare to https://github.com/mui/material-ui/blob/v3.x/packages/material-ui/src/Button/Button.js#L12